File: LayoutTests/http/tests/security/contentSecurityPolicy/report-same-origin-with-cookies-when-private-browsing-enabled.php
<?php
    header("Content-Security-Policy: img-src 'none'; report-uri /security/contentSecurityPolicy/resources/save-report.php");
?>
<!DOCTYPE html>
<html>
<body>
<script>
    if (window.testRunner)
        testRunner.setPrivateBrowsingEnabled(true);

    var xhr = new XMLHttpRequest();
    xhr.open("GET", "/cookies/resources/setCookies.cgi", false);
    xhr.setRequestHeader("SET-COOKIE", "hello=world;path=/");
    xhr.send(null);
</script>

<!-- This image will generate a CSP violation report. -->
<img src="/security/resources/abe.png">

<script src="resources/go-to-echo-report.js"></script>
</body>
</html>

File: LayoutTests/http/tests/security/contentSecurityPolicy/report-same-origin-with-cookies-when-private-browsing-enabled.py
#!/usr/bin/env python3

import sys

sys.stdout.write(
    'Content-Security-Policy: img-src \'none\'; report-uri /security/contentSecurityPolicy/resources/save-report.py\r\n'
    'Content-Type: text/html\r\n\r\n'
    '<!-- webkit-test-runner [ useEphemeralSession=true ] -->\n'
    '<!DOCTYPE html>\n'
    '<html>\n'
    '<body>\n'
    '<script>\n'
    '    var xhr = new XMLHttpRequest();\n'
    '    xhr.open("GET", "/cookies/resources/setCookies.cgi", false);\n'
    '    xhr.setRequestHeader("SET-COOKIE", "hello=world;path=/");\n'
    '    xhr.send(null);\n'
    '</script>\n'
    '\n'
    '<!-- This image will generate a CSP violation report. -->\n'
    '<img src="/security/resources/abe.png">\n'
    '\n'
    '<script src="resources/go-to-echo-report.js"></script>\n'
    '</body>\n'
    '</html>\n'
)